home/utils.py

```diff
@@ -204,37 +181,101 @@ def add_viber(row, page=None):
+                viber_body=get_body(viber_messages, "viber_message"),
                 viber_quick_replies=add_quick_replies(row["quick_replies"]),
                 locale=home_page.locale,
             )
         else:
             page.enable_viber = True
             page.viber_title = row["viber_title"]
-            page.viber_body = get_body(row, "viber_body", "viber_message")
+            page.viber_body = get_body(viber_messages, "viber_message")
             return page
 
+    def clean_row(row):
+        for field in (
+            "web_title",
+            "web_subtitle",
+            "web_body",
+            "whatsapp_title",
+            "whatsapp_body",
+            "next_prompt",
+            "viber_title",
+            "viber_body",
+            "messenger_title",
+            "messenger_body",
+            "parent",
+        ):
+            if field in row and row[field]:
+                row[field] = str(row[field]).strip()
+        return row
+
     if purge == "yes":
         ContentPage.objects.all().delete()
+        ContentPageIndex.objects.all().delete()
 
     if isinstance(locale, str):
         locale = Locale.objects.get(language_code=locale)
 
     home_page = HomePage.objects.get(locale__pk=locale.pk)
-    csv_file = file.read()
-    if isinstance(csv_file, bytes):
-        csv_file = csv_file.decode("utf-8")
-
-    reader = csv.DictReader(io.StringIO(csv_file))
-    for row in reader:
+    file = file.read()
+    lines = []
+    if filetype == "XLSX":
+        wb = load_workbook(filename=BytesIO(file))
+        ws = wb.active
+        ws.delete_cols(1, 7)
+        ws.delete_rows(1, 2)
+        for row in ws.iter_rows():
+            row_dict = {}
+            for index, x in enumerate(row):
+                row_dict[EXPORT_FIELDNAMES[index]] = x.value
+            lines.append(row_dict)
+    else:
+        if isinstance(file, bytes):
+            file = file.decode("utf-8")
+        reader = csv.DictReader(io.StringIO(file))
+        for dictionary in reader:
+            lines.append(dictionary)
+    for index, row in enumerate(lines):
+        if (
+            row["parent"] in ["", None]
+            and row["web_body"] in ["", None]
+            and row["whatsapp_body"] in ["", None]
+            and row["messenger_body"] in ["", None]
+        ):
+            cpi = ContentPageIndex(title=row["web_title"])
+            home_page.add_child(instance=cpi)
+            cpi.save_revision().publish()
+            continue
+        if row["web_title"] in ["", None]:
+            continue
         row = clean_row(row)
-        contentpage = add_web(row)
-        contentpage = add_whatsapp(row, contentpage)
-        contentpage = add_messenger(row, contentpage)
-        contentpage = add_viber(row, contentpage)
+        whatsapp_messages = [row["whatsapp_body"]]
+        messenger_messages = [row["messenger_body"]]
+        viber_messages = [row["viber_body"]]
+        counter = 1
+        while True:
```

Comment:
Maybe this could be `for next_row in lines[index + 1:]`, then we can remove the try/except